gems/smithy-client/sig/smithy-client/base.rbs

```diff
@@ -0,0 +1,8 @@
+module Smithy
+  module Client
+    # Base class for all service clients.
+    class Base
+      attr_reader config: Struct[untyped]
```

Comment:
Just as a general comment I think it's entirely reasonable to run rbs prototype on all the public classes and we can fill them in as we go.